### PR TITLE
Nt 106 create report post endpoint

### PR DESCRIPTION
**Pull request by fadestocodes**
## JIRA Ticket Link
https://atriadev.atlassian.net/jira/software/projects/NT/boards/38?selectedIssue=NT-106
## Migrations Required?

NO :negative_squared_cross_mark:

## Summary (a few sentences describing what this PR is doing)
I set up the url and View for reporting post to use /post/<int:pk>/report. This will serialize the request and response data, validate them, call the appropriate Service method, then return a HTTP status, message, and the reported_post object if successful. I also wrote a integration test to test a successful case as well as when a user_id is missing, and a post_id is non existent in DB.
## Checks

- [x] I have ran all the tests locally, and they all pass

## Impacted Areas in Application

List the areas in the app this PR is going to affect

- general backend

## Learnings

- getting more experience with writing integration tests using django testing library
- learned the workflows in django for serializing and validating data

### CODE REVIEW

File: townhall/posts/tests/test_report_post_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    def test_report_post_endpoint_missing_user_id(self):
+
+        # Arrange
+        test_data = {"post_id": self.test_post.pk}
+
+        # Act and Assert
+        response = self.client.post(self.url, test_data, format="json")
+        self.assertEqual(response.status_code, 400)
+
+    def test_report_post_endpoint_invalid_post(self):
```
Comment (ansel-rangers11) [inline townhall/posts/tests/test_report_post_endpoint.py:49]: Can we add a test for nonsensical data like None or post_id: "lol"

**Review by ansel-rangers11 — APPROVED** [review]
looks good, minor comment
